### PR TITLE
Added support for inputData

### PR DESCRIPTION
**Pull request by otopba**
https://github.com/vrtdev/flutter_workmanager/issues/67

### CODE REVIEW

File: android/src/main/kotlin/be/tramckrijte/workmanager/BackgroundWorker.kt [hunk 1]
```diff
@@ -34,8 +36,19 @@ class BackgroundWorker(private val ctx: Context,
...
 
-    private val dartTask
-        get() = workerParams.inputData.getString(DART_TASK_KEY)!!
+    private val inputData by lazy {
+        val map = HashMap<String, Any>()
+        for (entry in workerParams.inputData.keyValueMap.entries) {
+            val key = entry.key
+            when (val value = entry.value) {
+                //because io.flutter.plugin.common.StandardMessageCodec use Boolean.TRUE and Boolean.FALSE inside writeValue
+                is Boolean -> map[key] = if (value) java.lang.Boolean.TRUE else java.lang.Boolean.TRUE
```
Comment (timrijckaert) [inline android/src/main/kotlin/be/tramckrijte/workmanager/BackgroundWorker.kt]: this does not seem right

**Comment by timrijckaert** [comment]
Interesting. 
Thanks for this. Let me review your code a bit

File: android/src/main/kotlin/be/tramckrijte/workmanager/Extractor.kt [hunk 2]
```diff
@@ -269,4 +280,64 @@ object Extractor {
                 }
                 .build()
     }
+
+    private fun extractInputData(call: MethodCall): Map<String, Any>? {
```
Comment (timrijckaert) [inline android/src/main/kotlin/be/tramckrijte/workmanager/Extractor.kt]: Do we really need to _parse_ this map to a Kotlin Map on Android side?
Do we really care?
Can't we just store as an arbitrary `String` payload, but still return it as a `Map<String, dynamic>` on Dart side?

If so this method can be a whole lot simpler.

**Comment by otopba** [comment]
@timrijckaert Oh, good idea. Check it please

File: android/src/main/kotlin/be/tramckrijte/workmanager/Extractor.kt [hunk 3]
```diff
@@ -269,4 +278,13 @@ object Extractor {
...
                 .build()
     }
+
+    private fun extractPayload(call: MethodCall): String? {
+        try {
+            return call.argument<String>(REGISTER_TASK_PAYLOAD_KEY)
+        } catch (ex: Exception) {
+            Log.e(logTag, "Can't parse input data", ex)
+        }
+        return null
```
Comment (timrijckaert) [inline android/src/main/kotlin/be/tramckrijte/workmanager/Extractor.kt]: 🤔 Hmm.
Is this still relevant?

I think `call.argument<String>(REGISTER_TASK_PAYLOAD_KEY)` is enough.

File: lib/src/workmanager.dart [hunk 4]
```diff
@@ -73,8 +75,13 @@ class Workmanager {
   /// A helper function so you only need to implement a [BackgroundTaskHandler]
   static void executeTask(final BackgroundTaskHandler backgroundTask) {
     WidgetsFlutterBinding.ensureInitialized();
-    _backgroundChannel
-        .setMethodCallHandler((call) async => backgroundTask(call.arguments));
+    _backgroundChannel.setMethodCallHandler((call) async {
+      final payload = call.arguments["be.tramckrijte.workmanager.PAYLOAD"];
+      backgroundTask(
+        call.arguments["be.tramckrijte.workmanager.DART_TASK"],
+        payload == null ? null : jsonDecode(payload),
```
Comment (timrijckaert) [inline lib/src/workmanager.dart]: `payload ?? jsonDecode(payload)`

**Comment by timrijckaert** [comment]
Very close now 👍 
Thanks again for the work

File: lib/src/workmanager.dart (hunk 4 above)
Comment (otopba) [reply lib/src/workmanager.dart]: "??" Mean if null. But we need if NOT null

File: android/src/main/kotlin/be/tramckrijte/workmanager/Extractor.kt (hunk 3 above)
Comment (otopba) [reply android/src/main/kotlin/be/tramckrijte/workmanager/Extractor.kt]: Thank you. Removed

**Comment by otopba** [comment]
@timrijckaert Done. Check "payload ?? jsonDecode(payload)" please I leaved a comment

File: lib/src/workmanager.dart (hunk 4 above)
Comment (timrijckaert) [reply lib/src/workmanager.dart]: `payload == null ? null : jsonDecode(payload)`
if payload is `null` return `null` if it is not return the output of `jsonDecode(payload)`

`payload ?? jsonDecode(payload)`
if payload is null return it if it is not return the output of `jsonDecode(payload)` 

What am I missing?

Comment (otopba) [reply lib/src/workmanager.dart]: payload ?? jsonDecode(payload)
if payload is null return jsonDecode(payload) if it is not return it

image

Comment (timrijckaert) [reply lib/src/workmanager.dart]: right i see now. excuse me need some more sleep

Comment (otopba) [reply lib/src/workmanager.dart]: It is ok =)

**Pull request merged**